### DISCUSSION TITLE
`Filer`-less cross cluster continuous synchronization

### DISCUSSION THREAD

**Discussion by vobloeb** [seaweedfs/seaweedfs#3772] [Ideas]
`seafeedfs` can do [cross cluster continuous synchronization](https://github.com/seaweedfs/seaweedfs/wiki/Filer-Active-Active-cross-cluster-continuous-synchronization) following changes in `Filer`.

What would be the best way to set up cross cluster continuous synchronization if `Filer` is not going to be deployed ?

[Ambry](https://github.com/linkedin/ambry/wiki/Replication) does by making `volumeId`s globally-unique (like a `clusterA` has even ids and `clusterB` odd ids), so one cluster can subscribe to volume changes and pull volumes from another cluster as they got created and as they grow.
This does not require `Filer` and, unlike the current`Filer`-based synchronization, it is resilient to loops in replication topology.

`seafeedfs` seems superior by the features, but I see it lacks this one :(

**Comment by chrislusf** [comment]
Is this equivalent to using replication "100" and deploy to 2 data centers?

**Reply by vobloeb** [reply]
No.
I don't mean 1 multi-DC cluster, but 2+ clusters, each with its own `seaweed` `master` (or master quorum) so that both clusters are able to serve local clients (read and write) when the network is separated and there is no connection between the datacenters.

**Reply by chrislusf** [reply]
What is the use case with this?

**Reply by vobloeb** [reply]
I don't insist that `seaweedfs` supports this, it can be done by an external daemon.

My goal is to collect other people's user stories to see how they struggle with network splits and slowdowns between data centers.
